src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -518,7 +516,11 @@ private void RemovePinFromPoll(int pinNumber, ref int valueFileDescriptor, ref i
                 {
                     try
                     {
-                        _eventThreadCancellationTokenSource.Cancel();
+                        if (_eventThreadCancellationTokenSource != null)
```

Comment:
When could this be null? Is this just a check in case there are two different threads removing callbacks? If so, we should probably use a semaphore or lock instead of just a null check here.

Reply:
I had to add that condition because _eventThreadCancellationTokenSource is nullable after my changes. And I had to make _eventThreadCancellationTokenSource nullable because the "Nullable reference types" feature is enabled.

I had two options:
1. In the SysFsDriver constructor, I could assign a null with an exclamation mark at the end to point that I know what I'm doing when I left _eventThreadCancellationTokenSource no created. I mean this: 
`_eventThreadCancellationTokenSource = null!;`

2. I could make _eventThreadCancellationTokenSource nullable (this is what I did) and add new conditions to avoid compiler errors.
For example: In DetectEvents function, in the first while, I had to add _eventThreadCancellationTokenSource != null;
`while (_pinsToDetectEventsCount > 0 && _eventThreadCancellationTokenSource != null)`
I know _eventThreadCancellationTokenSource is not null because _pinsToDetectEventsCount is greather than 0, but the compiler wouldn't let me compile if I didn't show _eventThreadCancellationTokenSource was not null (adding the 
`_eventThreadCancellationTokenSource != null`)

Maybe the conditions make the project run more slowly?

Reply:
Ok, thanks for pointing out that this was due to nullability concerns. I think I'm okay with the changes you have except for one thing, which is the way you added this condition on the DetectEvents method. I will add a comment with my suggestion there, and explain why I believe we should make t hat change.

Reply:
In that while condition, I was worry about slow down the event detection thread. I never thought the Nullable reference types feature to let me add a condition before checking if _eventThreadCancellationTokenSource was null and go away from the function with an exception if it's true.

Yeah, I fully agree with your suggestion. And it solves my worry about the efficiency of the loop condition.